[c-pluff] Update to 0.1.4

hudokkow opened the pull request:
## Description

This updates the c-pluff lib from version 0.1.3 + patches to 0.1.4 + patches. I separated the patches and added them to patches/ dir. Should make it easier to update in the future. Most patches are small tweaks and some may not be needed anymore, with 2 notable exceptions: 0001-change-plugin-to-addon-tags.patch and 0003-missing-cpluff.dll-will-crash-Kod.patch
## Motivation and Context

I was curious to see how much our code differed from upstream c-pluff lib and it turns out it doesn't all that much. The exception is our favorite OS, Windows, with the biggest patch.
## How Has This Been Tested?

I've compile and shortly runtime tested this on linux. I've quickly tested installing add-ons and couldn't see anything weird on the debug log.
This will need a fair bit of runtime testing to see if everything still works.
## Screenshots (if appropriate):
## Types of change

<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Dependency update that, hopefully, doesn't break anything.
## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->

<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

For reference, I extracted lib/cpluff/ from core and placed it in https://github.com/hudokkow/kodi-cpluff with history preserved, including history from xbmc-antiquated tree.

c-pluff 0.1.4 already includes a slightly refactored version of https://github.com/hudokkow/kodi-cpluff/commit/1632bd2b3960250a2ebca0a940ccf2203e7a5878 written by JM.

A [jenkins run](http://jenkins.kodi.tv/view/Helpers/job/BuildMulti-All/1673/) showed that this introduced a problem with doxygen. I didn't noticed it on my machine because I have it installed. Uninstalled it and sure enough, build fails.

Odd thing is, if I build c-pluff standalone, it doesn't complain about missing doxygen. Autotools really isn't my thing and I couldn't find anything that might cause this in FindCpluff.cmake.
@wsnipex and @fetzerch, any ideas? Can you guys have a look?
The major change to how docs are built is https://github.com/jlehtine/c-pluff/commit/1673a38fcbb9431f91767fce34015fa3828f8986

Error is common to all failing builds:

> Making install in po
> Making install in test
> Making install in docsrc
> Making install in doc
> Making install in img
>  /bin/mkdir -p '/home/jenkins/workspace/LINUX-64/build/build/cpluff/share/doc/cpluff/html/img'
>  /usr/bin/install -c -m 644 architecture.png pageborder.png '/home/jenkins/workspace/LINUX-64/build/build/cpluff/share/doc/cpluff/html/img'
> /bin/sh: doxygen: command not found
> Makefile:880: recipe for target 'doc/reference/c-api/index.html' failed
> make[8]: **\* [doc/reference/c-api/index.html] Error 127
> Makefile:911: recipe for target 'doc' failed
> make[7]: **\* [doc] Error 2
> Makefile:710: recipe for target 'install-reference-doc' failed
> make[6]: **\* [install-reference-doc] Error 2
> Makefile:597: recipe for target 'install-am' failed
> make[5]: **\* [install-am] Error 2
> Makefile:436: recipe for target 'install-recursive' failed
> make[4]: **\* [install-recursive] Error 1
> Makefile:489: recipe for target 'install-recursive' failed
> make[3]: **\* [install-recursive] Error 1
> CMakeFiles/libcpluff.dir/build.make:74: recipe for target 'build/cpluff/src/libcpluff-stamp/libcpluff-install' failed
> make[2]: **\* [build/cpluff/src/libcpluff-stamp/libcpluff-install] Error 2
> CMakeFiles/Makefile2:447: recipe for target 'CMakeFiles/libcpluff.dir/all' failed
> make[1]: **\* [CMakeFiles/libcpluff.dir/all] Error 2
> Makefile:127: recipe for target 'all' failed
> make: **\* [all] Error 2
> Build step 'Execute shell' marked build as failure

For once win32 build went fine, specially after a change like this, which is quite surprising because I didn't updated the VS solution.
@Paxxi, can you check it?

hudokkow committed:
[c-pluff] Update to 0.1.4

hudokkow commented:
Its probably easier to see our patches at https://github.com/hudokkow/c-pluff/tree/0.1.x_kodi/patches

Also ping @Razzeee because [last attempt at bumping](https://github.com/xbmc/xbmc/pull/8753) has his.

akva2 commented:
this commit https://github.com/jlehtine/c-pluff/commit/449018d432f043fd03b806d33b55a0f954d2a0c0 should hint at why differences are rather small.

hudokkow commented:
> this commit jlehtine/c-pluff@449018d should hint at why differences are rather small.

I know. That's why i mentioned
> c-pluff 0.1.4 already includes a slightly refactored version of hudokkow/kodi-cpluff@1632bd2 written by JM.

win32 build for testers: http://mirrors.kodi.tv/test-builds/win32/KodiSetup-20161021-00d497a-cpluff_0.1.4.exe

Gave it a quick spin and all seems fine. Famous last words...

hudokkow commented:
jenkins build this please

hudokkow commented:
All builds successful and uploaded to mirrors. Just look for cpluff_0.1.4.

Paxxi commented:
Can we exclude the doc folder? Don't see much point in carrying php files in our repo

hudokkow commented:
Don't know. Are there any license implications on stripping a dir/component/whatever from a third-party work?

Paxxi commented:
I'm not a lawyer but it's under MIT license so as long as we save the license file and license headers in files it should be fine

MartijnKaijser commented:
Heck maybe make a reference to the original repo where to find the files? Also when it's moved to depends it's not really that important anymore to still include the /docs/ as you don't notice

Razzeee commented:
might be good to get this merged early in pre-alpha?

hudokkow commented:
Need to find the time to update, move to depends, test, etc.

afedchin commented:
Is there a plan to update windows dependencies?

Rechi commented:
@hudokkow can you handle the TEMP commit, so we can merge this

hudokkow commented:
Will get it done tonight.

hudokkow commented:
@afedchin, don't know. Unfortunately I'm a bit outside of the loop. Need to find some of those illusive 36h days. Or consider a career change... :disappointed:

@Rechi, rebased and moved `patches/` folder into a separate commit for easier review.
Thanks for keeping it rebased. :wink:

hudokkow commented:
Unbelievable! jenkins is green across the board... \o/

hudokkow commented on .gitignore:
```diff
@@ -144,39 +144,37 @@ cmake_install.cmake
+lib/cpluff/*/Makefile
+lib/cpluff/*/*/Makefile
+lib/cpluff/*/*/*/Makefile
+lib/cpluff/*/*/*/*/Makefile
```
Don't remember why. It was 1.5y ago. Probably something I read somewhere... SO perhaps...
@MartijnKaijser, sounds like a good idea after a 30s read.

hudokkow commented:
Fixed.

Rechi commented:
wrong revision was build
jenkins build this please

Rechi commented:
@hudokkow can you please add the following patch, otherwise we get a `implicit-function-declaration` warning

```
--- a/lib/cpluff/console/cmdinput_basic.c
+++ b/lib/cpluff/console/cmdinput_basic.c
@@ -26,8 +26,8 @@
 #include <stdio.h>
 #include <string.h>
+#include "console.h"
 #if defined(HAVE_ISATTY_FILENO)
 #include <unistd.h>
 #endif
-#include "console.h"
 
 #define CMDLINE_SIZE 256
```

hudokkow commented:
Will do tonight.

hudokkow commented:
@Rechi, done! Thanks and sorry for the 24h delay.

Rechi commented on lib/cpluff/patches/0003-win32-Fix-compile.-Missing-cpluff.dll-will-crash-Kod.patch:
```diff
@@ -0,0 +1,644 @@
++ * version of the C-Pluff distribution. Compile time software compatibility
++ * checks should use #CP_VERSION_MAJOR and #CP_VERSION_MINOR instead.
++ */
++#define CP_VERSION "0.1.3"
```
`0.1.4`
also needs to be updated in the real file

Rechi replied:
@hudokkow ping

hudokkow committed:
[c-pluff] Add patches folder for reference and keep track of changes
